components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultLogoutRequestHandler.java

```diff
@@ -152,6 +154,16 @@ protected void sendResponse(HttpServletRequest request, HttpServletResponse resp
         request.setAttribute(FrameworkConstants.ResponseParams.LOGGED_OUT, isLoggedOut);
 
         String redirectURL;
+        String webContextRoot = ServerConfiguration.getInstance().getFirstProperty(FrameworkConstants.WEB_CONTEXT_ROOT);
+        if (StringUtils.isNotBlank(webContextRoot)) {
+            if (webContextRoot.charAt(0) == '/') {
+                if (webContextRoot.length() <= 1) {
+                    webContextRoot = "";
+                }
+            } else {
+                webContextRoot = "/" + webContextRoot;
+            }
+        }
```

Comment:
Need to find out why we need these changes in authentication framework. Ideally they should all be part of the caller path variable send in the request.